common/app/routes/challenges/redux/selectors.js

```diff
@@ -36,14 +36,20 @@ const submitTypes = {
 export const challengeSelector = createSelector(
   state => state.challengesApp.challenge,
   state => state.entities.challenge,
-  (challengeName, challengeMap) => {
-    if (!challengeName || !challengeMap) {
+  state => state.entities.block,
+  (challengeName, challengeMap, blockMap) => {
+    if (
+      !challengeName ||
+      !challengeMap ||
+      !blockMap[challengeMap[challengeName].block]
```

Comment:
This is likely to throw error. Please add checks to make sure you are not attempting to read a prop from undefined